geomesa-process/geomesa-process-vector/src/main/scala/org/locationtech/geomesa/process/tube/TubeBuilder.scala

```diff
@@ -238,10 +238,10 @@ class InterpolatedGapFill(tubeFeatures: SimpleFeatureCollection,
         val dist = calc.getOrthodromicDistance
         //If the distance between points is greater than the buffer distance, segment the line
         //So that no segment is larger than the buffer. This ensures that each segment has an
-        //times and distance.
-        if (dist > bufferDistance) {
+        //times and distance. Also ensure that features do not share a time value.
+        val timeDiffMillis = t2.toInstant.toEpochMilli - t1.toInstant.toEpochMilli
+        if (dist > bufferDistance && timeDiffMillis > 0) {
           val heading = calc.getAzimuth
-          val timeDiffMillis = t2.toInstant.toEpochMilli - t1.toInstant.toEpochMilli
           val segCount = (dist / bufferDistance).toInt
           val segDuration = timeDiffMillis / segCount
```

Comment:
it would probably be good to validate the segDuration here - it seems like it could still be zero.

Reply:
on line 248 (which i can't comment on directly), it's also possible that the combination of the time difference and the segDuration will result in a single item list, which will cause an exception on line 257